File: 2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.py
class Solution:
    def minimumCost(self, cost: List[int]) -> int:
        cost.sort(reverse=True)
        n = len(cost)
        ans, x = 0, 0

        for i in range(n):
            if x < 2:
                ans += cost[i]
                x += 1
            else:
                x = 0

        return ans

File: 2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.cs
public class Solution {
    public int MinimumCost(int[] cost) {
        Array.Sort(cost);
        Array.Reverse(cost);
        
        int n = cost.Length;
        int ans = 0;
        
        for (int i = 0; i < n; i++) {
            if (i % 3 != 2) {
                ans += cost[i];
            }
        }
        
        return ans;
    }
}